File: 1524-string-matching-in-an-array/string-matching-in-an-array.py
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:


        # for i in range(len(words)):
        #     parent=words[i]
        #     for j in range(i+1,len(words)):
        words.sort(key=len)
        res=[]
        print(words)
        for i in range(len(words)):
            parent=words[i]
            for j in range(i+1,len(words)):
                if parent in words[j]:
                    res.append(parent)
                    break

        

        return res

        

File: 1524-string-matching-in-an-array/string-matching-in-an-array.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var stringMatching = function(words) {


    words.sort((a,b)=> a.length-b.length);
    let res=[];
    for(let i=0;i<words.length;i++)
    {
        let word = words[i];
        for(let j=i+1 ;j<words.length;j++)
        {
            if(words[j].includes(word))
            {
                res.push(word);
                break;

            }
        }
    }

    return res;

    
};